internal/resources/metal/project/resource.go

```diff
@@ -3,311 +3,304 @@ package project
-					validation.StringIsEmpty,
-				),
-			},
-			"organization_id": {
-				Type:        schema.TypeString,
-				Description: "The UUID of organization under which you want to create the project. If you leave it out, the project will be create under your the default organization of your account",
-				Optional:    true,
-				Computed:    true,
-				ForceNew:    true,
-				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
-					return strings.EqualFold(strings.Trim(old, `"`), strings.Trim(new, `"`))
-				},
-				ValidateFunc: validation.IsUUID,
-			},
-			"bgp_config": {
-				Type:        schema.TypeList,
-				Description: "Optional BGP settings. Refer to [Equinix Metal guide for BGP](https://metal.equinix.com/developers/docs/networking/local-global-bgp/)",
-				MaxItems:    1,
-				Optional:    true,
-				Elem: &schema.Resource{
-					Schema: map[string]*schema.Schema{
-						"deployment_type": {
-							Type:         schema.TypeString,
-							Description:  "\"local\" or \"global\", the local is likely to be usable immediately, the global will need to be review by Equinix Metal engineers",
-							Required:     true,
-							ValidateFunc: validation.StringInSlice([]string{"local", "global"}, false),
-						},
-						"asn": {
-							Type:        schema.TypeInt,
-							Description: "Autonomous System Number for local BGP deployment",
-							Required:    true,
-						},
-						"md5": {
-							Type:        schema.TypeString,
-							Description: "Password for BGP session in plaintext (not a checksum)",
-							Sensitive:   true,
-							Optional:    true,
-						},
-						"status": {
-							Type:        schema.TypeString,
-							Description: "Status of BGP configuration in the project",
-							Computed:    true,
-						},
-						"max_prefix": {
-							Type:        schema.TypeInt,
-							Description: "The maximum number of route filters allowed per server",
-							Computed:    true,
-						},
-					},
-				},
-			},
-		},
+		),
 	}
 }
 
-func expandBGPConfig(d *schema.ResourceData) (*metalv1.BgpConfigRequestInput, error) {
-	bgpDeploymentType, err := metalv1.NewBgpConfigRequestInputDeploymentTypeFromValue(d.Get("bgp_config.0.deployment_type").(string))
-	if err != nil {
-		return nil, err
-	}
-
-	bgpCreateRequest := metalv1.BgpConfigRequestInput{
-		DeploymentType: *bgpDeploymentType,
-		Asn:            int32(d.Get("bgp_config.0.asn").(int)),
-	}
-	md5, ok := d.GetOk("bgp_config.0.md5")
-	if ok {
-		bgpCreateRequest.Md5 = metalv1.PtrString(md5.(string))
-	}
+type Resource struct {
+	framework.BaseResource
+}
 
-	return &bgpCreateRequest, nil
+func (r *Resource) Schema(
+	ctx context.Context,
+	req resource.SchemaRequest,
+	resp *resource.SchemaResponse,
+) {
+	resp.Schema = resourceSchema(ctx)
 }
 
-func resourceMetalProjectCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	client := meta.(*config.Config).NewMetalClientForSDK(d)
+func (r *Resource) Create(
```

Comment:
Looks like something is missing here; haven't dug through this function yet but it looks like all tests that use a project resource are failing with this error:

```
=== NAME  TestAccDataSourceMetalPort_byId
    data_source_metal_port_acc_test.go:59: Step 1/1 error: Error running apply: exit status 1
        
        Error: Error creating project
        
          with equinix_metal_project.test,
          on terraform_plugin_test.tf line 109, in resource "equinix_metal_project" "test":
         109: resource "equinix_metal_project" "test" {
        
        Could not create project: API Error HTTP 404 404 Not Found
```

Reply:
it seems Im sending an empty organizationId when it is not defined